### PR TITLE
Feature/do-s3-storage

### LINKED ISSUE #276: Implement storage with S3 for Spaces
Digital Ocean Spaces has made itself compatible with the Amazon S3 SDK. We want to implement this to avoid having to write our own implementation.

### PR DESCRIPTION
**Pull request by tabeckers**
**What issue does this PR address?**
Fixes #276.

**Does this PR introduce a breaking change?**
Yes, the configuration for blob storage has changed.

### CODE REVIEW

File: src/FunderMaps.Infrastructure/Storage/FileStorageOptions.cs [hunk 1]
```diff
@@ -17,4 +23,4 @@ public sealed class FileStorageOptions
         /// </summary>
         public IDictionary<string, string> StorageContainers { get; set; } = new Dictionary<string, string>();
     }
-}
+}
```
Comment (yorickdewid) [inline src/FunderMaps.Infrastructure/Storage/FileStorageOptions.cs:26]: Newline

File: src/FunderMaps.Infrastructure/Storage/FileStorageOptions.cs [hunk 2]
```diff
@@ -2,9 +2,15 @@
 
 namespace FunderMaps.Infrastructure.Storage
```
Comment (yorickdewid) [inline src/FunderMaps.Infrastructure/Storage/FileStorageOptions.cs:3]: `#ifdef _DISABLED_ etc..`

File: src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        ///     Create new instance.
+        /// </summary>
+        public SpacesBlobStorageService(IOptions<BlobStorageOptions> options,
+            ILogger<SpacesBlobStorageService> logger)
+        {
+            if (options == null || options.Value == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+            options.Value.AccesKey.ThrowIfNullOrEmpty();
```
Comment (yorickdewid) [inline src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs:40]: Newline

File: src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            catch (Exception e)
+            {
+                // This type of exception indicates that the file does not exist.
+                if (e is AmazonS3Exception && ((AmazonS3Exception)e).ErrorCode == "NoSuchKey")
+                {
+                    return false;
+                }
+
+                _logger.LogError(e, "Could not check file existence in Spaces using S3");
+                // TODO QUESTION: Inner exception or not? I don't think so because we already log it.
```
Comment (yorickdewid) [inline src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs:96]: Include anyway

File: src/FunderMaps.WebApi/appsettings.json [hunk 5]
```diff
@@ -24,5 +24,11 @@
...
     "FunderMapsConnection": "",
     "AzureStorageConnectionString": ""
+  },
+  "BlobStorage": {
+    "AccesKey": "",
+    "SecretKey": "",
+    "ServiceUri": "",
+    "BlobStorageName": ""
   }
 }
```
Comment (yorickdewid) [inline src/FunderMaps.WebApi/appsettings.json:34]: Newline

File: src/FunderMaps.Webservice/Startup.cs [hunk 6]
```diff
@@ -80,7 +80,7 @@ public void ConfigureServices(IServiceCollection services)
             {
                 // FUTURE: The full enum description support for swagger with System.Text.Json is a WIP. This is a custom tempfix.
                 c.SchemaFilter<EnumSchemaFilter>();
-                c.GeneratePolymorphicSchemas();
+                c.GeneratePolymorphicSchemas(); // TODO Obsolete
```
Comment (yorickdewid) [inline src/FunderMaps.Webservice/Startup.cs:83]: I wrote the same comment in another branch

**Review by yorickdewid — CHANGES_REQUESTED** [review]

File: src/FunderMaps.Webservice/Startup.cs (hunk 6 above)
Comment (tabeckers) [reply src/FunderMaps.Webservice/Startup.cs:83]: We shall see my friend, we shall see

**Review by yorickdewid — APPROVED** [review]

**Review by yorickdewid — APPROVED** [review]

**Pull request merged**